magento/pwa-studio
# Added ability to use custom template for scaffolding.

## Description

This PR adds the ability to request the `scaffolding` program to use a custom template/version while creating a new project.

![image](https://user-images.githubusercontent.com/35203638/109327418-ee294980-781d-11eb-99ac-6f81a95a6f34.png)

![image](https://user-images.githubusercontent.com/35203638/109327524-0a2ceb00-781e-11eb-9443-4649a72c713f.png)

## Related Issue
Closes PWA-483

## Acceptance
Should be able to scaffold a new project using any compatible template on NPM.

### Verification Stakeholders
@jimbo @dpatil-magento 

### Specification
Need to update docs the mention the same.

### Verification Steps

Note: This won't work yet with `yarn create @magento/pwa` because we did not publish the new `@magento/create-pwa` yet. So let's test this using the new unpublished version locally.

1. Go to a folder where you would like to scaffold the new project.
2. In terminal run `node PATH_TO_PWA_STUDIO/packages/create-pwa/bin/create-pwa`
3. Answer all the questions.
4. For the question `Which template would you like to use to bootstrap test? Defaults to venia-concept.` enter any version from [NPM](https://www.npmjs.com/package/@magento/venia-concept). For instance: `@magento/venia-concept@9.0.1-alpha.3`. If you don't provide anything, it would check in the file system for `venia-concept`. If it can not find that, then it would search for a `venia-concept` in the local machine cache and use that instead.

## Checklist
* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
